src/lib/y2firewall/clients/auto.rb

```diff
@@ -62,28 +67,20 @@ def initialize
       #
       # @return [String]
       def summary
-        return Yast::HTML.Para(_("Firewalld is not available")) if !firewalld.installed?
-
-        firewalld.read if !firewalld.read?
-
-        # general overview
-        summary = general_summary
-
-        # per zone details
-        firewalld.zones.each do |zone|
-          summary << zone_summary(zone)
-        end
+        presenter = Y2Firewall::SummaryPresenter.new(firewalld)
+        return presenter.not_installed if !firewalld.installed?
+        return presenter.not_configured if !modified?
 
-        summary
+        presenter.create
       end
 
       # Import the firewall configuration
       #
       # @param profile [Hash] firewall profile section to be imported
       # @return [Boolean]
-      def import(profile)
+      def import(profile, merge = !Yast::Mode.config)
         self.class.profile = profile
-        return false unless read
+        return false if merge && !read(force: false)
```

Comment:
Hm, do you have tested this while a normal AY installation too ? With which configuration it will be merged ?
(It could depend on if we are in first or second stage)

Reply:
In a normal AutoYaST installation, we are always trying to merge the configuration with the current one avoiding to read it twice. 

Firewalld ships with a default configuration, predefined zones and services, so we read that configuration and just apply changes over it.

One of the problem with AutoYaST configuration is that we do some imports and exports when editing, so we need to avoid a read of the configuration in that cases, we only need to read when `Applying to the system`, and in this case we should remember the state before writing not showing the current system configuration but only the changes.

Reply:
@schubi2 
Tested it, also take into account that currently the firewall finish client does not support autoinstallation so, only called during a normal installation. That means that we are configuring the firewall only in the second stage.